connectors/sources/sharepoint_online.py

```diff
@@ -589,14 +589,7 @@ async def group_sites(self, group_id):
             return
 
     async def site_collections(self):
-        filter_ = url_encode("siteCollection/root ne null")
-        select = "siteCollection,webUrl"
-
-        async for page in self._graph_api_client.scroll(
-            f"{GRAPH_API_URL}/sites/?$filter={filter_}&$select={select}"
-        ):
-            for site_collection in page:
-                yield site_collection
+        yield await self._graph_api_client.fetch(url=f"{GRAPH_API_URL}/sites/root")
```

Comment:
https://learn.microsoft.com/en-us/graph/api/resources/site?view=graph-rest-1.0 says:

> /sites/root: The tenant root site.

and https://learn.microsoft.com/en-us/graph/api/site-list?view=graph-rest-1.0&tabs=http says:

> `siteCollection/root ne null` Lists all root-level site collections in the organization. Useful for discovering the home site for each geography.

Does the change proposed here mean that we'll lose the ability to handle geography-specific site roots? What happens for an organization that has multiple roots?

Reply:
Yes, the updated code fetches the root site of the tenant. Since, the only endpoint `/sites/root` works with the `Sites.Selected` permission, and other endpoint like `/sites/getAllSites` or `/sites?filter=siteCollections/root ne null` requires `Sites.Read.All` permission. 
If we don't have to change the endpoint, we can work around it as follows:
```
call /sites?filter=siteCollections/root ne null
if we get AccessDenied/Forbidden (403 status code)
then call /sites/root
``` 
This can prevent the connector to miss the geo-specific root sites. The connector fetches the sites according to the permission given. 
WDYT?

Reply:
👍 yeah, I like that approach better. This will protect us from any current users losing functionality that they rely on today. Thanks!